rstudio/rstudio
# Fix Inconsistent Emoji Spacing

### Intent

Visual mode renders text (including emoji) using a proportional font. This allows emoji to render at varying width, which can cause issues for certain emoji (e.g. the keycap emoji):

![Screen Shot 2020-11-20 at 8 28 25 AM](https://user-images.githubusercontent.com/261654/99811249-e5d0bc80-2b12-11eb-9627-677183193f9d.png)

This fix switches rendering to use of a fixed width font which should make the spacing consistent:

![Screen Shot 2020-11-20 at 8 28 44 AM](https://user-images.githubusercontent.com/261654/99811235-e0737200-2b12-11eb-9293-478f7cf448a7.png)


### Approach

The visual editor uses a fixed width font to render the emoji. In addition, I've updated other UI that displays emoji (completion, insert emoji dialog) to also use the fixed width font. Review the emoji rendering and exclude any emoji that don't render well using the new fixed width font (just a few).

### QA Notes

This fix should address rendering of emoji in visual mode as well as the supporting UI. It also switches the rendering of symbols in the insert symbol dialog to use a fixed width font (since the grid is shared between symbols and emoji). This shouldn't cause any issues, but if you notice anything strange with insert symbol, this could be related.
